core/io/image.cpp

```diff
@@ -3934,16 +3935,31 @@ Image::Image(const uint8_t *p_mem_png_jpg, int p_len) {
 	mipmaps = false;
 	format = FORMAT_L8;
 
-	if (_png_mem_loader_func) {
-		copy_internals_from(_png_mem_loader_func(p_mem_png_jpg, p_len));
+	if (p_len < 4) {
```

Comment:
You can delete length check, but all of the image format markers take 4 bytes. JPG marker also take 4 bytes, but last byte can be different for regular, EXIF and JFIF